### ISSUE TITLE
Taking gaussian mean action at the test time

### ISSUE THREAD

**Issue by tominku** [Unity-Technologies/ml-agents#1324]
Hi,

I want to only take gaussian mean action from the learned (ppo) policy, rather than sampled stochastic actions, in test (internal brain mode) time. Is there any way to do this ? 

Thanks.

**Comment by vincentpierre** [comment]
We do not have this feature in version 0.5. I would not recommend using the mean action only, it is better to train agents and carry inference the same way. You could end up with an agent stuck in an infinite loop if there is no stochasticity. Can I ask why you are trying to do this ?

**Comment by tominku** [comment]
In deterministic environments, the optimal action sequence might also be deterministic rather than stochastic. While the stochacity matters in training time for exploratory behavior, I think taking mean actions are more appropriate in testing the performance of the learned agent.

**Comment by vincentpierre** [comment]
In our next release, we will take the random normal variable outside of the model, so you might be able to modify the code in the internal brain then. In the mean time, you could modify the model at training time and multiply the random normal node by a placeholder variable (set it to 1 during training and to 0 at inference).

**Comment by harperj** [comment]
This discussion has been inactive for more than 30 days, so I'm going to close it.  Feel free to reopen if needed.

**Comment by ConorZAM** [comment]
How would this be done in V6? I'm having a similar issue.

**Comment by awjuliani** [comment]
Hi @ConorZAM 

This has been added in v0.6 and v0.7.

**Comment by ConorZAM** [comment]
Hey @awjuliani 

How would I do this? Is there a bool I can check in the discrete action applier class to bypass the multinomial part?

**Comment by awjuliani** [comment]
Hi @ConorZAM 

You will need to modify the inference brain script files. It is actually trickier for discrete actions, as they are represented as logits. In the gaussian case, the mean can be directly extracted however.
